js/app/controllers/calcontroller.js

```diff
@@ -140,17 +140,28 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 				$scope.$apply();
 			});
 		} else {
-			$scope.calendarsPromise = CalendarService.getPublicCalendar(constants.publicSharingToken).then(function(calendar) {
-				$scope.calendars = [calendar];
-				is.loading = false;
-				// TODO - scope.apply should not be necessary here
-				$scope.$apply();
-			}).catch((reason) => {
-				angular.element('#header-right').css('display', 'none');
-				angular.element('#emptycontent-container').css('display', 'block');
-			});
+		    $scope.calendars = [];
+		    var tokens = constants.publicSharingToken.split(".");
+		    var finished = 0;
+		    tokens.forEach(function(token){
+                $scope.calendarsPromise = CalendarService.getPublicCalendar(token).then(function(calendar) {
+                    finished++;
+                    $scope.calendars.push(calendar);
+                    if(finished >= tokens.length){
+                        is.loading = false;
+                        // TODO - scope.apply should not be necessary here
+                        $scope.$apply();
+                    }
+                }).catch((reason) => {
+                    angular.element('#header-right').css('display', 'none');
```

Comment:
This error handling needs to be adapted.
You don't want to show the empty-container error page if one out of ten calendars fail. 
You would want some kind of Notification though. You can use `OC.Notification.show()` for that.

Reply:
Will check.